src/main/java/com/marklogic/spark/writer/ArbitraryRowConverter.java

```diff
@@ -118,33 +138,29 @@ private ObjectNode readTree(String json) {
         }
     }
 
-    private String convertRowToJSONString(InternalRow row) {
-        StringWriter writer = new StringWriter();
-        JacksonGenerator jacksonGenerator = new JacksonGenerator(this.schema, writer, Util.DEFAULT_JSON_OPTIONS);
-        jacksonGenerator.write(row);
-        jacksonGenerator.flush();
-        return writer.toString();
-    }
-
     /**
      * jackson-xml-mapper unfortunately does not yet support a root namespace. Nor does it allow for the root element
      * to be omitted. So we always end up with "ObjectNode" as a root element. See
-     * https://github.com/FasterXML/jackson-dataformat-xml/issues/541 for more information.
-     * <p>
-     * While JSON-Java does not support a root namespace, it does allow for the root element to be omitted. That is
-     * sufficient for us, as we can then generate our own root element - albeit via string concatentation - that
-     * includes a user-defined namespace.
+     * https://github.com/FasterXML/jackson-dataformat-xml/issues/541 for more information. So this method does some
+     * work to replace that root element with one based on user inputs.
      *
-     * @param json
+     * @param doc
      * @return
      */
-    private String convertJsonToXml(String json) {
-        JSONObject jsonObject = new JSONObject(json);
-        if (this.xmlNamespace != null) {
-            StringBuilder xml = new StringBuilder(String.format("<%s xmlns='%s'>", this.xmlRootName, this.xmlNamespace));
-            xml.append(XML.toString(jsonObject, null));
-            return xml.append(String.format("</%s>", this.xmlRootName)).toString();
+    private String convertJsonToXml(JsonNode doc) {
+        try {
```

Comment:
Reworked to use Jackson XML. I'm not sure why I didn't do this in the first place - i.e. just replace the crappy "ObjectNode" root element with what the user wants.